### PR TITLE
feat: add grafana_silence module

### PR DESCRIPTION
**Pull request by flkhndlr**
##### SUMMARY

This PR adds the grafana_silence module, that is used to schedule, delete and manage silences.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

- grafana_silence
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/grafana_silence.py [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+#
+# Copyright: (c) 2023, flkhndlr (@flkhndlr)
+
+from __future__ import absolute_import, division, print_function
+
+DOCUMENTATION = """
+module: grafana_silence
+author:
+  - flkhndlr (@flkhndlr)
+version_added: "1.6.0"
```
Comment (Nemental) [inline plugins/modules/grafana_silence.py:26]:
```suggestion
version_added: "1.8.0"
```

File: plugins/modules/grafana_silence.py [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+            self._module.fail_json(
+                failed=True,
+                msg="Unauthorized to perform action '%s' on '%s'" % (method, full_url),
+            )
+        elif status_code == 403:
+            self._module.fail_json(failed=True, msg="Permission Denied")
+        elif status_code == 202:
+            return self._module.from_json(resp.read())
+        elif status_code == 200:
+            return self._module.from_json(resp.read())
```
Comment (Nemental) [inline plugins/modules/grafana_silence.py:231]:
```suggestion
        elif status_code in [200, 202]:
            return self._module.from_json(resp.read())
```

File: plugins/modules/grafana_silence.py [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+                msg="Silence with same parameters already exists! eg. '%s'"
+                % silence["id"],
+            )
+    elif state == "absent":
+        if silence:
+            grafana_iface.delete_silence(silence["id"])
+            changed = True
+        else:
+            module.exit_json(
+                failed=True,
```
Comment (rndmh3ro) [inline plugins/modules/grafana_silence.py:363]: IMO this should be `False`.
If a silence should be absent and it does not exist, then the desired target-state is reached, so everything is fine. Consider Ansibles `file`-module. If you delete a file with it, the module does not fail if the file is not there.

File: plugins/modules/grafana_silence.py [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+            )
+    elif state == "absent":
+        if silence:
+            grafana_iface.delete_silence(silence["id"])
+            changed = True
+        else:
+            module.exit_json(
+                failed=False,
+                changed=changed,
+                msg="No silence with the same parameters found! Deletion not possible",
```
Comment (rndmh3ro) [inline plugins/modules/grafana_silence.py:363]: To keep it in sync with the other modules, let's change this to:
```suggestion
                msg="Silence deleted.",
```

**Review by rndmh3ro — APPROVED** [review]

**Pull request merged**